packages/grant-explorer/src/features/round/ViewCartPage/SummaryContainer.tsx

```diff
@@ -0,0 +1,392 @@
+        setTimeout(() => {
+          navigate(`/`);
+        }, 5000);
+      }
+
+      if (
+        tokenApprovalStatus === ProgressStatus.IS_SUCCESS &&
+        voteStatus === ProgressStatus.IS_SUCCESS &&
+        txHash !== "" &&
+        !chainSwitched
+      ) {
+        setChainSwitched(true);
+        /** Get chains left to check out */
+        const chainsLeft = chainIdsBeingCheckedOut
+          .filter((x) => !checkedOutChainIds.includes(x))
+          .filter((id) => id !== currentChainId);
+
+        /** If we have succesfully checked out the last chain, redirect to thankyou */
+        if (chainsLeft.length === 0) {
+          setTimeout(() => {
+            setOpenMRCProgressModal(false);
+            // navigate(`/thankyou`);
+            console.log("SUCCESSSSSS");
+          }, modalDelayMs);
+          return;
+        }
+
+        /** Switch to the next chain */
+        const nextChain = chainsLeft[0];
+        await switchNetworkAsync?.(nextChain);
+
+        setCheckedOutChainIds((checkedOutChains) => {
+          return [...checkedOutChains, currentChainId];
+        });
+      }
+
+      if (chainSwitched && signer !== undefined) {
+        console.log("Post chain switch");
+        /*Chain has switched, fire off voting*/
+        const chainId =
+          ((await signer?.getChainId()) as ChainId) ??
+          chainIdsBeingCheckedOut[0];
+        setChainSwitched(false);
+        await submitDonations({
+          donations: projectsByChain[chainId],
+          donationToken: payoutTokens[chainId],
+          totalDonation: totalDonationsPerChain[chainId],
+          roundEndTime: currentPermitDeadline,
+          signer: signer as JsonRpcSigner,
+        });
+      }
+    }
+
+    run();
+  }, [tokenApprovalStatus, voteStatus, txHash, signer]);
+
+  const progressSteps = [
+    {
+      name: "Approve",
+      description: "Approve the contract to access your wallet",
+      status: tokenApprovalStatus,
+    },
+    {
+      name: "Submit",
+      description: "Finalize your contribution",
+      status: voteStatus,
+    },
+  ];
+
+  function handleConfirmation() {
+    // check to ensure all projects have donation amount
+    const emptyDonations = projects.filter(
+      (project) => !project.amount || Number(project.amount) === 0
+    );
+
+    if (emptyDonations.length > 0) {
+      setEmptyInput(true);
+      return;
+    }
+
+    // TODO: check if signer has enough token balance for the current round
+    // const accountBalance = rounds.find();
+    //
+    // if (!accountBalance || totalDonation.gt(accountBalance)) {
+    //   setInsufficientBalance(true);
+    //   return;
+    // }
+
+    setOpenChainConfirmationModal(true);
+  }
+
+  function PayoutModals() {
+    return (
+      <>
+        <ChainConfirmationModal
+          title={"Checkout"}
+          confirmButtonText={"Checkout"}
+          confirmButtonAction={handleSubmitDonation}
+          body={
```

Comment:
This would be more idiomatic as children

Reply:
Here it has both children and body, and as far as I can see they both serve a different purpose. Won't investigate why rn, not the OP of this code.